### PR TITLE
Migration to .Net Standard 2.0

### PR DESCRIPTION
**Pull request by EhRom**
The actual version of azure-apihub-sdk is no more compatible with Visual Studio new Functions model (in .Net Standard 2.0).

### CODE REVIEW

File: .nuget/Azure.ApiHub.Sdk.nuproj [hunk 1]
```diff
@@ -1,32 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (safihamid) [inline .nuget/Azure.ApiHub.Sdk.nuproj]: Thanks much for doing this! :-) We still need to deploy as Nuget . so we need corresponding nuspec files, ....

File: BuildPackage.cmd [hunk 2]
```diff
@@ -1,5 +0,0 @@
-@echo off
-
-REM nuget restore -NonInteractive
-
-msbuild Azure.ApiHub.Sdk.proj /t:Build /p:Configuration=Release
```
Comment (safihamid) [inline BuildPackage.cmd]: And a corresponding build file

File: .nuget/Azure.ApiHub.Sdk.nuproj (hunk 1 above)
Comment (fabiocav) [reply .nuget/Azure.ApiHub.Sdk.nuproj]: We shouldn't need nuspecs, but the proper build step using pack to create the packages.

File: samples/HttpTraces.txt [hunk 3]
```diff
@@ -1,1342 +0,0 @@
-PS G:\Temp> 
-"list apis"
-armclient get "/subscriptions/83e6374a-dfa5-428b-82ef-eab6c6bdd383/providers/Microsoft.Web/locations/brazilsouth/managedApis?api-version=2015-08-01-preview" -verbose
```
Comment (fabiocav) [inline samples/HttpTraces.txt]: @safihamid , is this something you no longer want to keep?

File: src/Azure.ApiHub.Sdk.csproj [hunk 4]
```diff
@@ -1,133 +1,36 @@
...
+<Project Sdk="Microsoft.NET.Sdk">
+
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{F8356D27-805D-4ECD-95BF-8CC08063EB02}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Microsoft.Azure.ApiHub.Sdk</RootNamespace>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (fabiocav) [inline src/Azure.ApiHub.Sdk.csproj]: We need a common prop/build properties file that is used by the project in this solution with the expected values (which will also end up in the package metadata). Please see an example here: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/3cfc5d2ce5e7f980861530dab71a9954485e7e69/build/common.props

And this is how this file is used: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/3cfc5d2ce5e7f980861530dab71a9954485e7e69/src/WebJobs.Extensions.Http/WebJobs.Extensions.Http.csproj#L2

Please also set the project specific properties (mainly for packaging) as it is done here: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/3cfc5d2ce5e7f980861530dab71a9954485e7e69/src/WebJobs.Extensions.Http/WebJobs.Extensions.Http.csproj#L3-L8

File: src/Azure.ApiHub.Sdk.csproj [hunk 5]
```diff
@@ -1,133 +1,36 @@
...
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <RunCodeAnalysis>true</RunCodeAnalysis>
+    <RootNamespace>Microsoft.Azure.ApiHub.Sdk</RootNamespace>
+    <PackageLicenseUrl>https://opensource.org/licenses/MIT</PackageLicenseUrl>
+    <PackageProjectUrl>https://github.com/EhRom/azure-apihub-sdk</PackageProjectUrl>
+    <AssemblyVersion>0.9.0.0</AssemblyVersion>
+    <FileVersion>0.9.0.0</FileVersion>
+    <Description>SDK for working with managed APIs and connections hosted in Azure App Service. Azure App Service platform abstracts different SaaS providers and makes easy to work with them with common contract.</Description>
+    <PackageReleaseNotes>Conversion to .Net Standard 2.0</PackageReleaseNotes>
```
Comment (fabiocav) [inline src/Azure.ApiHub.Sdk.csproj]: See the comment above. We need this properties to be correctly set (Company, Product, etc.)

File: src/Extensions/JsonExtensions.cs [hunk 6]
```diff
@@ -1,95 +1,93 @@
...
-        public static readonly JsonSerializer JsonMediaTypeSerializer = JsonSerializer.Create(JsonExtensions.MediaTypeFormatterSettings);
+        public static readonly JsonSerializer JsonObjectTypeSerializer = JsonSerializer.Create(ObjectSerializationSettings);
+        public static readonly JsonSerializer JsonMediaTypeSerializer = JsonSerializer.Create(MediaTypeFormatterSettings);
         public const int JsonSerializationMaxDepth = 512;
-        public static readonly MediaTypeFormatter JsonObjectTypeFormatter;
-        public static readonly MediaTypeFormatter[] JsonObjectTypeFormatters;
-        public static readonly MediaTypeFormatter JsonMediaTypeFormatter;
-        public static readonly MediaTypeFormatter[] JsonMediaTypeFormatters;
+
+        //public static readonly MediaTypeFormatter JsonObjectTypeFormatter;
```
Comment (fabiocav) [inline src/Extensions/JsonExtensions.cs]: If this code is no longer needed, we should remove it.

File: src/Extensions/JsonExtensions.cs [hunk 7]
```diff
@@ -1,95 +1,93 @@
...
-            };
-            System.Net.Http.Formatting.JsonMediaTypeFormatter mediaTypeFormatter2 = new System.Net.Http.Formatting.JsonMediaTypeFormatter();
-            mediaTypeFormatter2.SerializerSettings = JsonExtensions.MediaTypeFormatterSettings;
-            mediaTypeFormatter2.UseDataContractJsonSerializer = false;
-            JsonExtensions.JsonMediaTypeFormatter = (MediaTypeFormatter)mediaTypeFormatter2;
-            JsonExtensions.JsonMediaTypeFormatters = new MediaTypeFormatter[1]
-            {
-        JsonExtensions.JsonMediaTypeFormatter
-            };
+            //System.Runtime.Serialization.Formatter
```
Comment (fabiocav) [inline src/Extensions/JsonExtensions.cs]: Address commented out code. (If no longer needed, remove it)

File: test/Azure.ApiHub.Sdk.Tests/app.config [hunk 8]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (fabiocav) [inline test/Azure.ApiHub.Sdk.Tests/app.config]: Shouldn't need this or the redirect. As part of the migration, let's make sure the references are set to the versions used by the SDK.

File: src/Extensions/JsonExtensions.cs (hunk 6 above)
Comment (EhRom) [reply src/Extensions/JsonExtensions.cs]: I reactivated the code. As it was public I am not sure it was not used.

File: src/Azure.ApiHub.Sdk.csproj (hunk 4 above)
Comment (EhRom) [reply src/Azure.ApiHub.Sdk.csproj]: Changes are made in the last commit.

**Comment by EhRom** [comment]
I made changes to resolve two bugs (paramterless constructor in Azure CLI, and UriTemplate resolving). Do you know when the changes will be merged ?

File: src/Azure.ApiHub.Sdk/Azure.ApiHub.Sdk.csproj [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\..\build\common.props" />
```
Comment (safihamid) [inline src/Azure.ApiHub.Sdk/Azure.ApiHub.Sdk.csproj]: Are you missing the common.props file in your PR?

Comment (EhRom) [reply src/Azure.ApiHub.Sdk/Azure.ApiHub.Sdk.csproj]: Yes, you are right. The file was missing. I just added it to the repo.

**Comment by safihamid** [comment]
Thanks much! This is now merged to master. The v1 is moved to v1.x branch.

**Comment by EhRom** [comment]
Thanks! Is it planned to publish a nuget package (public or Azure Nightly) ? I have another pull request in order to merge these changes to the extensions repo : https://github.com/Azure/azure-webjobs-sdk-extensions/pull/358.

**Comment by safihamid** [comment]
Sure will do it today.

**Comment by safihamid** [comment]
This is published now.

**Pull request merged**